src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -184,14 +195,25 @@ public ProCombatMoveAI(final ProUtils utils, final ProBattleUtils battleUtils,
     // Calculate attack routes and perform moves
     doMove(attackMap, moveDel, data, player, isSimulation);
 
+    // Set strafing territories to avoid retreats
+    final List<Territory> strafingTerritories = new ArrayList<Territory>();
+    for (final Territory t : attackMap.keySet()) {
+      if (attackMap.get(t).isStrafing()) {
+        strafingTerritories.add(t);
+      }
+    }
+    ai.setStoredStrafingTerritories(strafingTerritories);
```

Comment:
Instead of having this be a simple setter, pass in the attackMap.keySet(). This then lets you simplify this long method and gives more functionality for the 'setSToredStrafingTerritories', you'll get to hid the for loop and the isStrafing check there, and get rid of the 'strafingTerritories' variable in this method altogether.